docker_deploy/docker_manager.go

```diff
@@ -279,13 +279,34 @@ func (dm *LocalDockerManager) PullImage(ctx context.Context, imageName string, r
 	return nil
 }
 
-func (dm *LocalDockerManager) CreateContainer(imageName string, repoDigest string, entry_point_args []string, options []string, logger logging.Logger, cancelCtx context.Context) (DockerContainer, error) {
+func (dm *LocalDockerManager) CreateContainer(imageName string, repoDigest string, entry_point_args []string, options map[string]interface{}, host_options map[string]interface{}, logger logging.Logger, cancelCtx context.Context) (DockerContainer, error) {
 	config := &container.Config{
 		Image: fmt.Sprintf("%s@%s", imageName, repoDigest),
 		Cmd:   entry_point_args,
 	}
 
-	resp, err := dm.dockerClient.ContainerCreate(cancelCtx, config, nil, nil, nil, "")
+	hostConfig := &container.HostConfig{}
+
+	for key, value := range host_options {
```

Comment:
unless i'm missing something, I don't see any tests that actually confirm the values are parsed and passed as expected

Reply:
yeah you're right.. added to validation.